src/cascadia/TerminalSettingsModel/IInheritable.h

```diff
@@ -206,3 +211,57 @@ private:                                                                    \
+    }                                                                       \
+                                                                            \
+    /* Returns the resolved value for this setting */                       \
+    /* fallback: user set value --> inherited value --> empty array */      \
+    std::array<type, size> name() const                                     \
+    {                                                                       \
+        const auto val{ _get##name##Impl() };                               \
+        return val ? *val : std::array<type, size>{};                       \
+    }                                                                       \
+                                                                            \
+    /* Overwrite the user set value */                                      \
+    void name(const std::array<type, size>& value)                          \
+    {                                                                       \
+        _##name = value;                                                    \
+    }                                                                       \
+                                                                            \
+    /* Clear the user set value */                                          \
+    void Clear##name()                                                      \
+    {                                                                       \
+        _##name = std::nullopt;                                             \
+    }                                                                       \
+                                                                            \
+private:                                                                    \
+    std::optional<std::array<type, size>> _##name{ std::nullopt };          \
+    std::optional<std::array<type, size>> _get##name##Impl() const          \
+    {                                                                       \
+        /*return user set value*/                                           \
+        if (_##name)                                                        \
+        {                                                                   \
+            return _##name;                                                 \
+        }                                                                   \
+                                                                            \
+        /*user set value was not set*/                                      \
+        /*iterate through parents to find a value*/                         \
+        for (auto parent : _parents)                                        \
+        {                                                                   \
+            if (auto val{ parent->_get##name##Impl() })                     \

```

Comment:
Specifically this bit I'm worried about - are we returning a _copy_ of the parent's result?